packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -0,0 +1,137 @@
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import { BaseControllerV2 } from '@metamask/base-controller';
+
+const controllerName = 'QueuedRequestController';
+
+export const QueuedRequestControllerActionTypes = {
+  enqueueRequest: `${controllerName}:enqueueRequest` as const,
+};
+
+export const QueuedRequestControllerEventTypes = {
+  countChanged: `${controllerName}:countChanged` as const,
+};
+
+export type QueuedRequestControllerState = Record<string, never>;
+
+export type QueuedRequestControllerCountChangedEvent = {
+  type: typeof QueuedRequestControllerEventTypes.countChanged;
+  payload: [number];
+};
+
+export type QueuedRequestControllerEnqueueRequestAction = {
+  type: typeof QueuedRequestControllerActionTypes.enqueueRequest;
+  handler: QueuedRequestController['enqueueRequest'];
+};
+
+export type QueuedRequestControllerEvents =
+  QueuedRequestControllerCountChangedEvent;
+
+export type QueuedRequestControllerActions =
+  QueuedRequestControllerEnqueueRequestAction;
+
+export type QueuedRequestControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  QueuedRequestControllerActions,
```

Comment:
Typing fix to follow pattern established in other controllers.

```suggestion

type AllowedActions =
  | NetworkControllerGetStateAction
  | NetworkControllerSetActiveNetworkAction
  | NetworkControllerSetProviderTypeAction
  | NetworkControllerGetNetworkClientByIdAction
  | NetworkControllerFindNetworkClientIdByChainIdAction
  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
  | AddApprovalRequest;

export type QueuedRequestControllerMessenger = RestrictedControllerMessenger<
  typeof controllerName,
  QueuedRequestControllerActions | AllowedActions,
```

Reply:
Required changes in import statements
```diff
@@ -1,5 +1,14 @@
+import type { AddApprovalRequest } from '@metamask/approval-controller';
 import type { RestrictedControllerMessenger } from '@metamask/base-controller';
 import { BaseControllerV2 } from '@metamask/base-controller';
+import type {
+  NetworkControllerFindNetworkClientIdByChainIdAction,
+  NetworkControllerGetNetworkClientByIdAction,
+  NetworkControllerGetStateAction,
+  NetworkControllerSetActiveNetworkAction,
+  NetworkControllerSetProviderTypeAction,
+} from '@metamask/network-controller';
+import type { SelectedNetworkControllerSetNetworkClientIdForDomainAction } from '@metamask/selected-network-controller';
```